### PR TITLE
Add Feature: restore

### PR DESCRIPTION
**Pull request by Zehui-Lin**

### CODE REVIEW

**Comment by fadel** [comment]
Hi Zehui-Lin, thanks for your pull request. I will have time to evaluate it next week. For now I have the following in my mind.

It looks good to me as is, but I am wondering if we could achieve a similar feature without changing the behavior of `copy_to()`. The reasoning behind this being that for large models, users might be concerned about unintentional copies of parameters.

Maybe a `store()` method for collecting the params to be restored later with `.restore()`?

**Comment by Zehui-Lin** [comment]
Good suggestion!
👍

File: torch_ema/ema.py [hunk 1]
```diff
@@ -60,16 +60,30 @@ def copy_to(self, parameters):
             if param.requires_grad:
                 param.data.copy_(s_param.data)
 
+    def store(self, parameters):
```
Comment (fadel) [inline torch_ema/ema.py:62]: Really nice! Could you also remove lines 57 and 59 (those that reference `self.collected_parameters` in `copy_to`)?

Other than that, I'd also suggest adding `self.collected_parameters = []` in `__init__()` and I'd be ready to merge.

**Review by fadel — CHANGES_REQUESTED** [review]
Sorry for taking so long to get back to you. I added one minor comment. Thanks for your contribution!

File: torch_ema/ema.py (hunk 1 above)
Comment (Zehui-Lin) [reply torch_ema/ema.py:62]: Of course.

**Pull request merged**